### PR TITLE
Support Dir.glob with Pathname.

### PR DESCRIPTION
**Pull request by craigw**
This works with stdlib:

    require 'pathname'
    p = Pathname.new('/')
    Dir[p]
    # => ["/"]

But before this change failed under MemFs:

    TypeError:
      no implicit conversion of Pathname into String

This is more likely to be encountered in a Rails project where
`Rails.root` is a `Pathname`, but it's very possible to encounter this
in an Ruby project.

### CODE REVIEW

File: spec/memfs/dir_spec.rb [hunk 1]
```diff
@@ -216,6 +217,8 @@ module MemFs
       it_behaves_like 'returning matching filenames', '/test?', %w[/test0 /test1 /test2]
       it_behaves_like 'returning matching filenames', '/test[01]', %w[/test0 /test1]
       it_behaves_like 'returning matching filenames', '/test[^2]', %w[/test0 /test1]
+      it_behaves_like 'returning matching filenames', Pathname.new('/'), %w[/]
+      it_behaves_like 'returning matching filenames', Pathname.new('/*'), %w[/tmp /test0 /test1 /test2]
```
Comment (simonc) [inline spec/memfs/dir_spec.rb]: Codacy Issue found: [Line is too long. [103/80]](https://www.codacy.com/app/happynoff/memfs/file/2685713311/issues/source?bid=3240728&fileBranchId=3336523#l221)

**Comment by craigw** [comment]
Hmm I don't think Travis is supporting Ruby older than 2.0.0 any more. It fails wit this message:

> Gem::InstallError: ruby_dep requires Ruby version >= 2.0.0, ~> 2.0.

File: spec/memfs/dir_spec.rb [hunk 2]
```diff
@@ -216,6 +217,9 @@ module MemFs
       it_behaves_like 'returning matching filenames', '/test?', %w[/test0 /test1 /test2]
       it_behaves_like 'returning matching filenames', '/test[01]', %w[/test0 /test1]
       it_behaves_like 'returning matching filenames', '/test[^2]', %w[/test0 /test1]
+      it_behaves_like 'returning matching filenames', Pathname.new('/'), %w[/]
+      it_behaves_like 'returning matching filenames', Pathname.new('/*'),
+        %w[/tmp /test0 /test1 /test2]
```
Comment (simonc) [inline spec/memfs/dir_spec.rb]: Codacy Issue found: [Align the parameters of a method call if they span more than one line.](https://www.codacy.com/app/happynoff/memfs/file/2685727877/issues/source?bid=3240728&fileBranchId=3336523#l222)

**Comment by simonc** [comment]
Hi @craigw. Thank you very much for your PR that's so cool! 😁

**Pull request closed without merging**